tools/ext4dist.py

```diff
@@ -174,7 +174,16 @@
 b = BPF(text=bpf_text)
 
 # Common file functions. See earlier comment about generic_file_read_iter().
-b.attach_kprobe(event="generic_file_read_iter", fn_name="trace_read_entry")
+# Comment by Joe Yin 
+# From Linux 4.10, the function .read_iter at the ext4_file_operations has 
+# changed to ext4_file_read_iter.
+# So, I add the try and except,it will first to attach ext4_file_read_iter,
+# if fails and will attach the generic_file_read_iter which used to pre-4.10.
+
+try:
+	b.attach_kprobe(event="ext4_file_read_iter", fn_name="trace_entry")
```

Comment:
Maybe you can use BPF helper `get_kprobe_functions`? This way, we do not need try-and-error approach. Several other tools uses `get_kprobe_functions` as well.